### ISSUE TITLE
Simplify and unify golang agents dependencies structure

### ISSUE THREAD

**Issue by garagatyi** [eclipse-che/che#4464]
We use some golang libraries for our golang based agents. we moved some of them several times and now structure is not very clean.
We should:
1. Vendor [this commit of gorilla/websocket ](https://github.com/gorilla/websocket/commit/361d4c0ffd78338ebe0a9e6320cdbe115d7dc026) instead of using [this dependency](https://github.com/eclipse/che-lib/tree/master/websocket).
1. Remove [this lib](https://github.com/eclipse/che-lib/tree/master/websocket-terminal) because it is already vendored in the main Che repo.
1. Vendor [this commit of kr/pty](https://github.com/kr/pty/commit/5cf931ef8f76dccd0910001d74a58a7fca84a83d) instead of using [this dependency](https://github.com/eclipse/che-lib/tree/master/pty) and patch it with [commit 1](https://github.com/codenvy/pty/commit/af8b7524bf739cbe1abe8ecb9b1867c233186060) and [commit 2](https://github.com/codenvy/pty/commit/ab685d7210182dca053fc90de10cf23fd251b5a5)
1. Remove folders `pty, websocket-terminal, websocket` from [Che lib](https://github.com/eclipse/che-lib)

**Comment by garagatyi** [comment]
To clarify here is what we did:
1. We forked dependencies into che-lib.
1. We applied patches.
1. We moved agents code into Che.
1. We vendored dependencies forked into che-lib.

Here is what we should do:
1. Rework vendoring config to use origin repos with commit IDs that were used from the beginning in  che-lib.
1. Apply patches that were aplied in che-lib.
1. Remove forked dependencies from che-lib.
